client/src/components/Posting.tsx

```diff
@@ -0,0 +1,54 @@
+import { StarIcon } from "@chakra-ui/icons";
+import { Badge, Box } from "@chakra-ui/react";
+import { useNavigate } from "react-router-dom";
+import { Post } from "./interfaces/Post";
+
+export interface props {
+  post: Post;
+}
+
+const Posting: React.FC<props> = (props) => {
```

Comment:
```suggestion
const Posting = (props: { post: Post }): JSX.Element => {
```
No need to export the props interface